Rename DispatchKey::UndefinedTensorId to Undefined

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32788 Add XLAPreAutograd key for XLA use cases that need custom autograd.
* #32787 Stop using dispatchTypeId to do checks for tensor list unwrap.
* #32734 Centralize addition of "always on" dispatch keys.
* #32729 Make DispatchKeyGuards accept DispatchKey::Undefined
* **#32728 Rename DispatchKey::UndefinedTensorId to Undefined**

It doesn't have much to do with tensors anymore.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19628093](https://our.internmc.facebook.com/intern/diff/D19628093)

ezyang committed:
Rename DispatchKey::UndefinedTensorId to Undefined

It doesn't have much to do with tensors anymore.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 35230c2a:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=35230c2a5ae6909ba297dec4506d51c9c9b7b999).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4366556) pytorch_linux_xenial_cuda10_1_cudnn7_py3_NO_AVX_NO_AVX2_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=92419026) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=92419026)) :snowflake:
<details>
<summary>
<code>Jan 29 02:56:41 RuntimeError: quantization scale should be > 0</code>
</summary>

```
Jan 29 02:56:41   File "test_quantization.py", line 730, in test_quantized_rnn 
Jan 29 02:56:41     y, (h, c) = cell_dq(x, (h, c)) 
Jan 29 02:56:41   File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 540, in __call__ 
Jan 29 02:56:41     result = self.forward(*input, **kwargs) 
Jan 29 02:56:41   File "/opt/conda/lib/python3.6/site-packages/torch/nn/quantized/dynamic/modules/rnn.py", line 373, in forward 
Jan 29 02:56:41     return self.forward_tensor(input, hx) 
Jan 29 02:56:41   File "/opt/conda/lib/python3.6/site-packages/torch/nn/quantized/dynamic/modules/rnn.py", line 334, in forward_tensor 
Jan 29 02:56:41     input, hx, batch_sizes, max_batch_size, sorted_indices) 
Jan 29 02:56:41   File "/opt/conda/lib/python3.6/site-packages/torch/nn/quantized/dynamic/modules/rnn.py", line 315, in forward_impl 
Jan 29 02:56:41     self.batch_first, dtype=self.dtype, use_dynamic=True) 
Jan 29 02:56:41 RuntimeError: quantization scale should be > 0 
Jan 29 02:56:41  
Jan 29 02:56:41 ---------------------------------------------------------------------- 
Jan 29 02:56:41 Ran 36 tests in 60.131s 
Jan 29 02:56:41  
Jan 29 02:56:41 FAILED (errors=1, skipped=1) 
Jan 29 02:56:41  
Jan 29 02:56:41 Generating XML reports... 
Jan 29 02:56:41 Traceback (most recent call last): 
Jan 29 02:56:41   File "test/run_test.py", line 456, in <module> 
Jan 29 02:56:41     main() 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

ezyang committed:
Update on "Rename DispatchKey::UndefinedTensorId to Undefined"


It doesn't have much to do with tensors anymore.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19628093](https://our.internmc.facebook.com/intern/diff/D19628093)

[ghstack-poisoned]

ezyang committed:
Update on "Rename DispatchKey::UndefinedTensorId to Undefined"


It doesn't have much to do with tensors anymore.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19628093](https://our.internmc.facebook.com/intern/diff/D19628093)

[ghstack-poisoned]
